Code/Tools/TestImpactFramework/Runtime/Common/Code/Include/Static/BuildTarget/Common/TestImpactBuildGraph.h

```diff
@@ -59,8 +77,35 @@ namespace TestImpact
         //! Returns the vertex for the specified build target, else throws `BuildTargetException`.
         const BuildGraphVertex<ProductionTarget, TestTarget>& GetVertexOrThrow(
             const BuildTarget<ProductionTarget, TestTarget>& buildTarget) const;
-            
+
+        //! Walks the specified target's build dependencies.
+        void WalkBuildDependencies(
+            const BuildTarget<ProductionTarget, TestTarget>& buildTarget,
+            const BuildGraphVertexVisitor<ProductionTarget, TestTarget>& visitor) const;
+
+        //! Walks the specified target's build dependers.
+        void WalkBuildDependers(
+            const BuildTarget<ProductionTarget, TestTarget>& buildTarget,
+            const BuildGraphVertexVisitor<ProductionTarget, TestTarget>& visitor) const;
+
+        //! Walks the specified target's runtime dependencies.
+        void WalkRuntimeDependencies(
+            const BuildTarget<ProductionTarget, TestTarget>& buildTarget,
+            const BuildGraphVertexVisitor<ProductionTarget, TestTarget>& visitor) const;
+
+        //! Walks the specified target's runtime dependers.
+        void WalkRuntimeDependers(
+            const BuildTarget<ProductionTarget, TestTarget>& buildTarget,
+            const BuildGraphVertexVisitor<ProductionTarget, TestTarget>& visitor) const;
+
     private:
+        //! Generic function for walking the specified target's build grpah.
+        void WalkTargetBuildGraphSet(
+            TargetBuildGraph<ProductionTarget, TestTarget> BuildGraphVertex<ProductionTarget, TestTarget>::*graph,
+            TargetBuildGraphSet<ProductionTarget, TestTarget> TargetBuildGraph<ProductionTarget, TestTarget>::*set,
```

Comment:
It's a bit of a funky syntax but it's a (pointer to data member)[https://docs.microsoft.com/en-us/cpp/cpp/pointer-to-member-operators-dot-star-and-star?view=msvc-170]. It's useful because it means we can have the same generic function to handle the build and runtime dependencies and dependers. I'll add a brief note explaining it 👍

Reply:
A Type Alias can be used to make it look better if you know 
```
using BuildGraphSetPointer = TargetBuildGraphSet<ProductionTarget, TestTarget> TargetBuildGraph<ProductionTarget, TestTarget>::*;
```

Reply:
Good suggestion. Fixed 👍